### PR TITLE
[dart-dio] fix endpoints returning null

### PR DESCRIPTION
**Pull request by ahmednfwela**
Modify mustache files to handle cases where the response data is null (e.g. status code 204) instead of throwing an exception
fixes https://github.com/OpenAPITools/openapi-generator/issues/15426

### PR checklist
 
- [x] Read the [contribution guidelines](https://github.com/openapitools/openapi-generator/blob/master/CONTRIBUTING.md).
- [x] Pull Request title clearly describes the work in the pull request and Pull Request description provides details about how to validate the work. Missing information here may result in delayed response from the community.
- [x] Run the following to [build the project](https://github.com/OpenAPITools/openapi-generator#14---build-projects) and update samples:
  ```
  ./mvnw clean package 
  ./bin/generate-samples.sh
  ./bin/utils/export_docs_generators.sh
  ``` 
  Commit all changed files. 
  This is important, as CI jobs will verify _all_ generator outputs of your HEAD commit as it would merge with master. 
  These must match the expectations made by your contribution. 
  You may regenerate an individual generator by passing the relevant config(s) as an argument to the script, for example `./bin/generate-samples.sh bin/configs/java*`. 
  For Windows users, please run the script in [Git BASH](https://gitforwindows.org/).
- [ ] In case you are adding a new generator, run the following additional script : 
  ```
  ./bin/utils/ensure-up-to-date.sh
  ``` 
  Commit all changed files.
- [x] File the PR against the [correct branch](https://github.com/OpenAPITools/openapi-generator/wiki/Git-Branches): `master` (6.3.0) (minor release - breaking changes with fallbacks), `7.0.x` (breaking changes without fallbacks)
- [x] If your PR is targeting a particular programming language, @mention the [technical committee](https://github.com/openapitools/openapi-generator/#62---openapi-generator-technical-committee) members, so they are more likely to review the pull request.

@jaumard (2018/09) @josh-burton (2019/12) @amondnet (2019/12) @sbu-WBT (2020/12) @kuhnroyal (2020/12) @agilob (2020/12) @ahmednfwela (2021/08)

### CODE REVIEW

**Review by agilob — APPROVED** [review]

**Comment by ahmednfwela** [comment]
@wing328

File: modules/openapi-generator/src/main/resources/dart/libraries/dio/api.mustache [hunk 1]
```diff
@@ -116,7 +116,7 @@ class {{classname}} {
     );
     {{#returnType}}
 
-    {{{returnType}}} _responseData;
+    {{{.}}}? _responseData;
```
Comment (kuhnroyal) [inline modules/openapi-generator/src/main/resources/dart/libraries/dio/api.mustache:119]: Don't we know if the response may be null? Maybe we can also change the operation generic return type?

Comment (ahmednfwela) [reply modules/openapi-generator/src/main/resources/dart/libraries/dio/api.mustache:119]: it won't matter since the final response.data is nullable anyway

Comment (kuhnroyal) [reply modules/openapi-generator/src/main/resources/dart/libraries/dio/api.mustache:119]: True, that's a dio probelm.

**Review by kuhnroyal — APPROVED** [review]

**Comment by AyushBherwani1998** [comment]
@wing328 @ahmednfwela if we should allow developers to define whether they want the response to be nullable or not, data being nullable adds lots of check for the null value. For some usecases, if the data is null, we may want to throw an exception. It should be customisable IMO. Your thoughts?

**Comment by ahmednfwela** [comment]
having the response nullable suits all cases, and then devs can throw exceptions them selves if the response is null

**Pull request merged**